Orange/tests/test_misc.py

```diff
@@ -28,3 +33,60 @@ def test_memoize_method(self):
         self.assertEqual(calc.my_sum(1, 2, 3, 4, 5), 15)
         # Make sure different args produce different results
         self.assertEqual(calc.my_sum(1, 2, 3, 4), 10)
+
+
+class TestFlaggedData(unittest.TestCase):
+    def setUp(self):
+        Variable._clear_all_caches()
+        self.zoo = Table("zoo")
+
+    def test_create_flagged_table(self):
+        flagged = create_flagged_table(self.zoo, list(range(10)))
+
+        # check flagged table domain
+        self.assertEqual(flagged.domain.variables, self.zoo.domain.variables)
+        self.assertEqual(2, len(flagged.domain.metas))
+        self.assertIn(self.zoo.domain.metas[0], flagged.domain.metas)
+        self.assertIn(FLAGGED_FEATURE_NAME, [m.name for m in flagged.domain.metas])
+
+        # check flagged table data
+        np.testing.assert_array_equal(flagged.X, self.zoo.X)
+        np.testing.assert_array_equal(flagged.Y, self.zoo.Y)
+        np.testing.assert_array_equal(flagged.metas[:, 0].ravel(),
+                                      self.zoo.metas.ravel())
+        self.assertEqual(10, np.sum([i[FLAGGED_FEATURE_NAME] for i in flagged]))
+
+    def test_create_flagged_table_selected(self):
+        # check flagged column for no selected indices
+        flagged = create_flagged_table(self.zoo, [])
+        self.assertEqual(len(flagged), len(self.zoo))
+        self.assertEqual(0, np.sum([i[FLAGGED_FEATURE_NAME] for i in flagged]))
+
+        # check flagged column fol all selectes indices
+        flagged = create_flagged_table(self.zoo, list(range(len(self.zoo))))
+        self.assertEqual(len(flagged), len(self.zoo))
+        self.assertEqual(len(self.zoo),
+                         np.sum([i[FLAGGED_FEATURE_NAME] for i in flagged]))
+
+    def test_create_flagged_table_none_data(self):
+        self.assertIsNone(create_flagged_table(None, None))
+
+    def test_create_flagged_table_none_indices(self):
+        flagged = create_flagged_table(self.zoo, None)
+        self.assertEqual(len(flagged), len(self.zoo))
+        self.assertEqual(0, np.sum([i[FLAGGED_FEATURE_NAME] for i in flagged]))
+
+    def test_cascade_flagged_tables(self):
```

Comment:
I could be too smart by half and replace your function that uses regular expressions with one that uses the first unoccupied name. This test wouldn't fail because there are no "holes". Can you add some code to this test to remove the second meta and then "flag" the table again, so my smart idea would fail?